pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -23,20 +23,22 @@ import (
 )
 
 var (
-	resyncPeriod = 1 * time.Minute
-	apiVersion   = fmt.Sprintf("%s/%s", v1alpha1.GroupVersion.Group, v1alpha1.GroupVersion.Version)
+	resyncPeriod                    = 1 * time.Minute
+	apiVersion                      = fmt.Sprintf("%s/%s", v1alpha1.GroupVersion.Group, v1alpha1.GroupVersion.Version)
+	featurFlagConfigurationResource = v1alpha1.GroupVersion.WithResource("featureflagconfigurations")
 )
 
 type Sync struct {
-	Logger       *logger.Logger
-	ProviderArgs sync.ProviderArgs
-	URI          string
-
-	ready      bool
-	namespace  string
-	crdName    string
-	readClient client.Reader
-	informer   cache.SharedInformer
+	Logger        *logger.Logger
+	ProviderArgs  sync.ProviderArgs
+	URI           string
+	ReadClient    client.Reader
```

Comment:
What's the reason for exporting this field?

Reply:
I think refactoring the code a bit and introducing a `NewK8sSync(read, dynamic)` function we don't need to expose the two clients. At the moment they are exposed to construct the object in the runtime package